test/TestProjects/MgmtDiscriminator/Generated/MgmtDiscriminatorModelFactory.cs

```diff
@@ -0,0 +1,317 @@
+        /// </param>
+        /// <param name="actions">
+        /// A list of actions that are executed when all the conditions of a rule are satisfied.
+        /// Please note <see cref="DeliveryRuleAction"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="DeliveryRuleCacheExpirationAction"/>, <see cref="DeliveryRuleCacheKeyQueryStringAction"/>, <see cref="DeliveryRuleRequestHeaderAction"/>, <see cref="DeliveryRuleResponseHeaderAction"/>, <see cref="OriginGroupOverrideAction"/>, <see cref="DeliveryRuleRouteConfigurationOverrideAction"/>, <see cref="UrlRedirectAction"/>, <see cref="UrlRewriteAction"/> and <see cref="UrlSigningAction"/>.
+        /// </param>
+        /// <param name="extraMappingInfo">
+        /// A dictionary of mapping details about the actions that are executed when all the conditions of a rule are satisfied.
+        /// Please note <see cref="DeliveryRuleAction"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="DeliveryRuleCacheExpirationAction"/>, <see cref="DeliveryRuleCacheKeyQueryStringAction"/>, <see cref="DeliveryRuleRequestHeaderAction"/>, <see cref="DeliveryRuleResponseHeaderAction"/>, <see cref="OriginGroupOverrideAction"/>, <see cref="DeliveryRuleRouteConfigurationOverrideAction"/>, <see cref="UrlRedirectAction"/>, <see cref="UrlRewriteAction"/> and <see cref="UrlSigningAction"/>.
+        /// </param>
+        /// <returns> A new <see cref="Models.DeliveryRuleProperties"/> instance for mocking. </returns>
+        public static DeliveryRuleProperties DeliveryRuleProperties(int? order = null, DeliveryRuleCondition conditions = null, IEnumerable<DeliveryRuleAction> actions = null, IDictionary<string, DeliveryRuleAction> extraMappingInfo = null)
+        {
+            actions ??= new List<DeliveryRuleAction>();
+            extraMappingInfo ??= new Dictionary<string, DeliveryRuleAction>();
+
+            return new DeliveryRuleProperties(order, conditions, actions?.ToList(), extraMappingInfo);
+        }
+
+        /// <summary> Initializes a new instance of DeliveryRuleCondition. </summary>
+        /// <param name="name"> The name of the condition for the delivery rule. </param>
+        /// <param name="foo"> For test. </param>
+        /// <returns> A new <see cref="Models.DeliveryRuleCondition"/> instance for mocking. </returns>
+        public static DeliveryRuleCondition DeliveryRuleCondition(string name = null, string foo = null)
+        {
+            return new UnknownDeliveryRuleCondition(name, foo);
```

Comment:
The discriminator here should default to "Unknown" if its an extensible enum or default(EnumType) if its a fixed enum.

Do we have a case for the fixed enum discriminator?

Reply:
I wonder if we have a case that the discriminator is a fixed enum because fixed enums mean it could never expand, and if it never expands, what is the point of "Unknown"?

As for the default value, let me investigate if we could change it to Unknown.
But we cannot use Unknown enum value because we do not have one, here we can only use the Unknown literal.

Reply:
The default value for discriminators is `Unknown` if it appears in the method signature and it only appears in method signature when this is the base class